### PR TITLE
update readme and support kind to deploy kosmos

### LINKED ISSUE #365: Project documentation is inadequate and technical implementation descriptions are not detailed enough
**What would you like to be added**:
1. add quick start, contact, contributing, how to build to readme
2. add some technical  image and description for clustertree module

**Why is this needed**:
1. readme Lack of code compilation instructions, how to contribute code and participate in the project
2. clustertree technical documentation needs further improvement

### PR DESCRIPTION
**Pull request by village-way**
#### What type of PR is this?
/kind cleanup

#### What does this PR do?

1. update readme and fix kosmosctl version can not overwrite by make VERSION env param
2. add dev workflow images for local debug and test
3. update readme, delete Deprecated serviceAccount config, update make clean
4. Add config for ci.yml to enable manual triggering
5. macos make improve, add go clean cache(clean.sh), add verify-all.sh for make verify
6. add support to deploy kosmos by kind

#### Which issue(s) does this PR fix?

Fixes #365 
#### Special notes for your reviewer:
can not make on macos

@duanmengkk 
@assign duanmengkk
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: docs/clustertree/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# ClusterTree(集群容器编排)
```
Comment (ONE7live) [inline docs/clustertree/README.md:1]: Is it okay to put this doc here? what do you think @duanmengkk

File: .github/workflows/dev_images.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        uses: "WyriHaximus/github-action-get-previous-tag@v1"
+        with:
+          fallback: 1.0.0
+      - name: Push images
+        env:
+          ON_PLUGINS: true
+          REGISTRY: ghcr.io/${{ github.repository_owner }}/${{ github.repository_name }}
+          TAG: ${{ steps.previoustag.outputs.tag }}
+        run: |
+          make upload-images REGISTRY=${REGISTRY%?} VERSION=$TAG
```
Comment (duanmengkk) [inline .github/workflows/dev_images.yml:33]: Why do we need this?

File: Makefile [hunk 3]
```diff
@@ -9,6 +9,15 @@ REGISTRY_PASSWORD?=""
 REGISTRY_SERVER_ADDRESS?=""
 KIND_IMAGE_TAG?="v1.25.3"
 
+MACOS_TARGETS := clusterlink-controller-manager  \
+				kosmos-operator \
+				 clusterlink-elector \
+				clusterlink-network-manager \
+				clusterlink-proxy \
+				clustertree-cluster-manager \
+				scheduler \
```
Comment (duanmengkk) [inline Makefile:18]: We can use `make mp-image-clusterlink-controller-manager ` to make arm images

File: README.md [hunk 4]
```diff
@@ -45,24 +45,105 @@ The Kosmos scheduling module is an extension developed on top of the Kubernetes
...
 kosmosctl join cluster --name cluster1 --kubeconfig ~/kubeconfig/cluster1-kubeconfig  --cni calico --default-nic eth0  --enable-all
 kosmosctl join cluster --name cluster2 --kubeconfig ~/kubeconfig/cluster2-kubeconfig  --cni calico --default-nic eth0  --enable-all
 ```
-And then we can Use the Kosmos clusters like single cluster.
+##### 5. Use the Kosmos clusters like single cluster on the control plane.
+```shell
+kubectl get nodes
+```
+- use this command we can see the two cluster nodes Name like `kosmos-*` roles as `agent`
+<div><img src="docs/images/demo-check-node.png" style="width:400px;"  alt="nodes"/></div>
```
Comment (duanmengkk) [inline README.md:134]: Seens good，But we need to redesign the entire document.

File: docs/clustertree/README.md (hunk 1 above)
Comment (duanmengkk) [reply docs/clustertree/README.md:1]: We need to redesign the entire document.

**Comment by duanmengkk** [comment]
This is a nice code submission, but this PR contains too many features. We need to break it down, so I will close it for now. /cc @village-way 
/close

File: .github/workflows/dev_images.yml (hunk 2 above)
Comment (village-way) [reply .github/workflows/dev_images.yml:33]: to push dev image to person ghcr

**Pull request closed without merging**